tests/test_workers/test_tasks/test_build.py

```diff
@@ -807,6 +807,130 @@ def side_effect(_, temp_dir, *args, **kwargs):
         mock_dep_b.assert_not_called()
 
 
+@mock.patch('iib.workers.tasks.build.deprecate_bundles')
+@mock.patch('iib.workers.tasks.utils.get_resolved_bundles')
+@mock.patch('iib.workers.tasks.build._cleanup')
+@mock.patch('iib.workers.tasks.build._verify_labels')
+@mock.patch('iib.workers.tasks.build._prepare_request_for_build')
+@mock.patch('iib.workers.tasks.build._update_index_image_build_state')
+@mock.patch('iib.workers.tasks.build._opm_index_add')
+@mock.patch('iib.workers.tasks.build._build_image')
+@mock.patch('iib.workers.tasks.build._push_image')
+@mock.patch('iib.workers.tasks.build._verify_index_image')
+@mock.patch('iib.workers.tasks.build._update_index_image_pull_spec')
+@mock.patch('iib.workers.tasks.build.export_legacy_packages')
+@mock.patch('iib.workers.tasks.build.set_request_state')
+@mock.patch('iib.workers.tasks.build._create_and_push_manifest_list')
+@mock.patch('iib.workers.tasks.build.get_legacy_support_packages')
+@mock.patch('iib.workers.tasks.build.validate_legacy_params_and_config')
+@mock.patch('iib.workers.tasks.build.gate_bundles')
+@mock.patch('iib.workers.tasks.build.get_resolved_bundles')
+@mock.patch('iib.workers.tasks.build._add_label_to_index')
+@mock.patch('iib.workers.tasks.build._get_present_bundles')
+@mock.patch('iib.workers.tasks.build.set_registry_token')
+def test_handle_add_request_check_index_label_behavior(
+    mock_srt,
+    mock_gpb,
+    mock_alti,
+    mock_grb,
+    mock_gb,
+    mock_vlpc,
+    mock_glsp,
+    mock_capml,
+    mock_srs,
+    mock_elp,
+    mock_uiips,
+    mock_vii,
+    mock_pi,
+    mock_bi,
+    mock_oia,
+    mock_uiibs,
+    mock_prfb,
+    mock_vl,
+    mock_cleanup,
+    mock_ugrb,
+    mock_dep_b,
+):
+    arches = {'amd64', 's390x'}
+    binary_image_config = {'prod': {'v4.5': 'some_image'}}
+    mock_prfb.return_value = {
+        'arches': arches,
+        'binary_image': 'binary-image:latest',
+        'binary_image_resolved': 'binary-image@sha256:abcdef',
+        'from_index_resolved': 'from-index@sha256:bcdefg',
+        'ocp_version': 'v4.5',
+        'distribution_scope': 'stage',
+    }
+    mock_grb.return_value = ['some-bundle@sha', 'some-deprecation-bundle@sha']
+    legacy_packages = {'some_package'}
+    mock_glsp.return_value = legacy_packages
+    output_pull_spec = 'quay.io/namespace/some-image:3'
+    mock_capml.return_value = output_pull_spec
+    mock_gpb.return_value = [{'bundlePath': 'random_bundle@sha'}], ['random_bundle@sha']
+    bundles = ['some-bundle:2.3-1', 'some-deprecation-bundle:1.1-1']
+    cnr_token = 'token'
+    organization = 'org'
+    greenwave_config = {'some_key': 'other_value'}
+    mock_ugrb.return_value = ['random_bundle@sha', 'some-deprecation-bundle@sha']
+    deprecation_list = ['random_bundle@sha', 'some-deprecation-bundle@sha']
+    # Assume default labels are set on the index
+    os.environ['LABEL_SET'] = 'default_labels_set'
```

Comment:
Why use `os.environ` here? 

You can create a variable here and it will be available in the functions below. You just have to make sure to use a mutable data type, e.g. `dict`:
```python
label_state = {'LABEL_SET': 'default_labels_set'}
```

Then you can update `label_state['LABEL_SET']` as much as you want. It will always change the same `dict` object.

Reply:
you're right. A mutable data type can be used indeed.